common/lib/client/auth.js

```diff
@@ -97,7 +108,33 @@ var Auth = (function() {
 			Logger.logAction(Logger.LOG_ERROR, 'Auth()', msg);
 			throw new Error(msg);
 		}
-	}
+	};
+
+	/**
+	 * Update the authentication options for this client;
+	 * this will also update these options on the realtime connection
+	 * so that they take effect immediately
+	 * @param options
+	 */
+	Auth.prototype.setOptions = function(options) {
```

Comment:
> authorise() just forces the generation of a new token using current options.

Not according to the spec.  See [RSA10b](http://docs.ably.io/client-lib-development-guide/features/#RSA10b) and [RSA10g](http://docs.ably.io/client-lib-development-guide/features/#RSA10g).

> setOptions sets the options on which authorise() acts

Why not just call `authorise` then?

> What if you are setting a token explicitly in options ? 

Why not do it in authorise, `client.auth.authorise({ token: token })`?

> What if you change the token capabilities by specifying different params in the authURL ?

What's the problem. Calling `authorise` will issue the request again with any params configured in `authorise({}, { authParams: { "newParam": "value" }, force: true })`

> You can't change either of those things just by calling authorise().

Yes you can